src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberFieldSymbol.cs

```diff
@@ -420,34 +420,10 @@ internal override bool HasPointerType
         {
             get
             {
-                if (_lazyTypeAndRefKind?.Type.DefaultType is { } defaultType)
-                {
-                    bool isPointerType = defaultType.Kind switch
-                    {
-                        SymbolKind.PointerType => true,
-                        SymbolKind.FunctionPointerType => true,
-                        _ => false
-                    };
-                    Debug.Assert(isPointerType == IsPointerFieldSyntactically());
-                    return isPointerType;
-                }
-
-                return IsPointerFieldSyntactically();
+                return TypeWithAnnotations.DefaultType.IsPointerOrFunctionPointer();
```

Comment:
i both made these changes to not be syntactic (as requested), and i tried to add tests to see if i could cause any infnite-loops/circularities.  I couldn't find any problems.  That said, if someone could point me at existing struct circularity tests, that would be helpful so i can potential brainstorm more complicated cases to throw at this code to ensure there's no issue.

Reply:
There's a comment in `BaseTypeAnalysis` which we should look at more:
```
        // NOTE: If we do not check HasPointerType, we will unconditionally
        //       bind Type and that may cause infinite recursion.
        //       HasPointerType can use syntax directly and break recursion.
        internal static TypeSymbol NonPointerType(this FieldSymbol field) =>
            field.HasPointerType ? null : field.Type;
```

Reply:
done.